packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -365,11 +378,12 @@ class IOSDevice extends Device {
   String get _buildVersion => _getDeviceInfo(id, 'BuildVersion');
 
   @override
-  DeviceLogReader get logReader {
-    if (_logReader == null)
-      _logReader = new _IOSDeviceLogReader(this);
+  DeviceLogReader get logReader => logReaderForApp(null);
 
-    return _logReader;
+  @override
+  DeviceLogReader logReaderForApp(ApplicationPackage app) {
+    _logReaders ??= <ApplicationPackage, _IOSDeviceLogReader>{};
```

Comment:
Yeah, it occurred to me that almost everything in flutter_tools lives as long as the process and so this isn't any different.

My long term dream would be for flutter_tools to actually live a long time and operate like a daemon that clients speak to, at that point it may matter. I don't think we need to address this in this CL but maybe add a TODO where we add to the maps to someday remove from them too.